models/metabayesian_MH.jl

```diff
@@ -0,0 +1,48 @@
+#=
+This is a "meta-Bayesian" model, where the generative model includes an inversion of a different generative model.
+These types of models are common in cognitive modelling, where systems of interest (e.g. human subjects) are thought to use Bayesian inference to navigate their environment.
+Here we use a Metropolis-Hasting sampler implemented with Turing as the inversion of the inner "subjective" model.
+=#
+
+# Inner model function
+@model function inner_model(observation, prior_μ = 0, prior_σ = 1)
+    
+    # The innter model's prior
+    mean ~ Normal(prior_μ, prior_σ)
+
+    # The inner model's likelihood
+    observation ~ Normal(mean, 1)
+end
+
+# Outer model function
+@model function metabayesian_MH(observation, action, inner_sampler = MH(), inner_n_samples = 20)
+    
+    ### Sample parameters for the inner inference and response ###
+
+    #The inner model's prior's sufficient statistics
+    subj_prior_μ ~ Normal(0, 1)
+    subj_prior_σ = 1.0 
+    
+    # #Inverse temperature for actions
+    β ~ 1
```

Comment:
```suggestion
    # #Inverse temperature for actions
    β = 1
```

This is probably why this model is failing. I'll also fix that on #37.

Reply:
It actually did run locally for me with estimating the beta (just took ages). But I removed it for simplicity. 
Let me know if you need anything :)

Reply:
Would you prefer to keep it as beta ~ Exponential(1)? I could try changing it back to that.

Reply:
No the simplicity is better I suppose - what matters is that it can differentiate through the Turing call :) 
Noise parameters like that are common (ubiquitous) in these types of models, but not necessary for testing the differentiation :)